### PR TITLE
Add support for fetching SBOMs

### PR DESCRIPTION
**Pull request by vandanrohatgi**
Closes https://github.com/google/go-github/issues/2864

Adds support for fetching Software Bill of materials (SBOM) for a repository.

- [x] go generate
- [x] go test
- [x] go vet

### CODE REVIEW

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:16]: "SBOM" should be all caps since it is an acronym... here, and throughout this PR.

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:24]: We don't use `*time.Time` in this repo, we use `*Timestamp`.

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:25]: This should be `[]string`, not `[]*string`.

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:22]:
```suggestion
// CreationInfo represents when the SBOM created and who created it.
```

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:28]:
```suggestion
// RepoDependencies represents the dependencies of a repo.
type RepoDependencies struct {
```

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:29]:
```suggestion
	SPDXID *string `json:"SPDXID,omitempty"`
```

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:42]:
```suggestion
// SBOMInfo represents a software bill of materials (SBOM) using SPDX.
// SPDX is an open standard for SBOMs that
// identifies and catalogs components, licenses, copyrights, security
// references, and other metadata relating to software.
type SBOMInfo struct {
```

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:44]:
```suggestion
	SPDXID       *string       `json:"SPDXID,omitempty"`
	SPDXVersion  *string       `json:"spdxVersion,omitempty"`
```

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:50]:
```suggestion
	DocumentDescribes []string `json:"documentDescribes,omitempty"`
```

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:61]:
```suggestion
// GetSBOM fetches the software bill of materials for a repository.
```

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:64]:
```suggestion
func (s *DependencyGraphService) GetSBOM(ctx context.Context, owner, repo string) (*SBOM, *Response, error) {
```

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:19]:
```suggestion
	SBOM *SBOMInfo `json:"sbom,omitempty"`
```

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:72]:
```suggestion
	var sbom *SBOM
```

File: github/github.go
Comment (gmlewis) [inline github/github.go:210]: Please do not delete this line.

File: github/github.go
Comment (gmlewis) [inline github/github.go:350]: Please do not delete this line.

**Review by gmlewis — CHANGES_REQUESTED** [review]

File: github/dependency_graph.go
Comment (vandanrohatgi) [reply github/dependency_graph.go:16]: This change applies to comments as well as the code(such as function name) right?

File: github/dependency_graph.go
Comment (vandanrohatgi) [reply github/dependency_graph.go:25]: I'm new to Go. Could you please explain why?

Comment (vandanrohatgi) [reply github/dependency_graph.go:25]: Might as well club all my questions, I saw usage of %v when formatting the API URI instead of %s like this:

`fmt.Sprintf("repos/%v/%v/dependency-graph/sbom", owner, repo)`

Is there a reason for this?

Comment (gmlewis) [reply github/dependency_graph.go]: https://github.com/google/go-github/issues/2563#issuecomment-1313753582

Pointers are used for structs and are not needed for strings.

`%v` is typically used in idiomatic Go unless there is a good reason not to, such as when passing a `[]byte`.

Comment (vandanrohatgi) [reply github/dependency_graph.go]: Understood. Thank you!

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:15]:
```suggestion
// SBOM represents a software bill of materials, which descibes the
```

File: github/dependency_graph.go
Comment (gmlewis) [inline github/dependency_graph.go:21]:
```suggestion
// CreationInfo represents when the SBOM was created and who created it.
```

**Review by gmlewis — CHANGES_REQUESTED** [review]
Thank you, @vandanrohatgi !
Just a couple minor comment nits, please, that I apparently missed the first time
before  we await a second LGTM+Approval from any other contributor to this repo before merging.

**Comment by vandanrohatgi** [comment]
Thank you so much for the valuable feedback @gmlewis ! Learned a lot. Though I feel I only increased your workload rather than contributing :)

**Comment by gmlewis** [comment]
> Thank you so much for the valuable feedback @gmlewis ! Learned a lot. Though I feel I only increased your workload rather than contributing :)

No, not at all, @vandanrohatgi !  Your contributions are appreciated more than you know!
I couldn't maintain this repo without the help of wonderful contributors like yourself.

**Review by gmlewis — APPROVED** [review]
Thank you, @vandanrohatgi !
LGTM.

Awaiting second LGTM+Approval from any other contributor to this repo before merging.

**Comment by gmlewis** [comment]
@Parker77 - do you have time for a code review, please?

File: github/dependency_graph.go
Comment (Parker77) [inline github/dependency_graph.go:15]:
```suggestion
// SBOM represents a software bill of materials, which describes the
```

**Review by Parker77 — APPROVED** [review]
LGTM!

**Comment by gmlewis** [comment]
Thank you, @Parker77 !
Merging.

**Pull request merged**
